src/hooks/users.ts

```diff
@@ -1,52 +1,61 @@
 import { useMemo } from "react";
-import { useFirestoreConnect } from "react-redux-firebase";
+import { isLoaded } from "react-redux-firebase";
 
 import { usersSelector, usersByIdSelector } from "utils/selectors";
 import { WithId } from "utils/id";
 
 import { useSelector } from "hooks/useSelector";
 import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
 import { useVenueId } from "hooks/useVenueId";
+import { useSparkleFirestoreConnect } from "hooks/useSparkleFirestoreConnect";
 
 import { User } from "types/User";
 
 const useConnectUsers = () => {
-  useFirestoreConnect("users");
+  const venueId = useVenueId();
+
+  useSparkleFirestoreConnect([
+    {
+      collection: "users",
+      where: ["enteredVenueIds", "array-contains", venueId],
+      storeAs: "users",
+    },
+  ]);
 };
 
-export const usePartygoers = () => {
+export const usePartygoers = (): readonly WithId<User>[] => {
   useConnectUsers();
 
   const lastSeenThreshold = useUserLastSeenThreshold();
 
-  const users = useSelector(usersSelector);
+  const users = useSelector(usersSelector) ?? [];
 
   return useMemo(
     () => users.filter((user) => user.lastSeenAt > lastSeenThreshold),
     [users, lastSeenThreshold]
   );
```

Comment:
Following on from https://github.com/sparkletown/sparkle/pull/1042#discussion_r551731447

In the old `useCampPartygoers` we had:

```
partygoers.filter(
  (partygoer) => partygoer?.lastSeenIn?.[venueName] > lastSeenThreshold
),
```

Whereas in the new `usePartygoers` with your latest changes, we have:

```typescript
users.filter((user) => user.lastSeenAt > lastSeenThreshold)
```

What is the reasoning for using `lastSeenAt` as opposed to `lastSeenIn?.[venueName]` here? I think the old way is more correct, but i'm interested to hear your reasoning.

Reply:
See also https://github.com/sparkletown/internal-sparkle-issues/issues/119